hibernate-reactive-core/src/main/java/org/hibernate/reactive/loader/entity/ReactiveCacheEntityLoaderHelper.java

```diff
@@ -0,0 +1,148 @@
+			final LoadEventListener.LoadType options) throws HibernateException {
+
+		SessionImplementor session = event.getSession();
+		Object old = session.getEntityUsingInterceptor( keyToLoad );
+
+		if ( old != null ) {
+			// this object was already loaded
+			EntityEntry oldEntry = session.getPersistenceContext().getEntry( old );
+			if ( options.isCheckDeleted() ) {
+				if ( oldEntry.getStatus().isDeletedOrGone() ) {
+					LoadingLogger.LOGGER.debug(
+							"Load request found matching entity in context, but it is scheduled for removal; returning null" );
+					return new PersistenceContextEntry( old, EntityStatus.REMOVED_ENTITY_MARKER );
+				}
+			}
+			if ( options.isAllowNulls() ) {
+				final EntityPersister persister = event.getSession()
+						.getFactory()
+						.getRuntimeMetamodels()
+						.getMappingMetamodel()
+						.getEntityDescriptor( keyToLoad.getEntityName() );
+				if ( !persister.isInstance( old ) ) {
+					LOG.debug(
+							"Load request found matching entity in context, but the matched entity was of an inconsistent return type; returning null"
+					);
+					return new PersistenceContextEntry( old, EntityStatus.INCONSISTENT_RTN_CLASS_MARKER );
+				}
+			}
+			upgradeLock( old, oldEntry, event.getLockOptions(), event.getSession() );
+		}
+
+		return new PersistenceContextEntry( old, EntityStatus.MANAGED );
+	}
+
+	/**
+	 * see org.hibernate.event.internal.AbstractLockUpgradeEventListener#upgradeLock(Object, EntityEntry, LockOptions, EventSource)
+	 */
+	private static void upgradeLock(Object object, EntityEntry entry, LockOptions lockOptions, EventSource source) {
+
+		LockMode requestedLockMode = lockOptions.getLockMode();
+		if ( requestedLockMode.greaterThan( entry.getLockMode() ) ) {
+			// The user requested a "greater" (i.e. more restrictive) form of pessimistic lock
+
+			if ( entry.getStatus() != Status.MANAGED ) {
+				throw new ObjectDeletedException(
+						"attempted to lock a deleted instance",
+						entry.getId(),
+						entry.getPersister().getEntityName()
+				);
+			}
+
+			final EntityPersister persister = entry.getPersister();
+			final boolean cachingEnabled = persister.canWriteToCache();
+			SoftLock lock = null;
+			Object ck = null;
+			try {
+				if ( cachingEnabled ) {
+					EntityDataAccess cache = persister.getCacheAccessStrategy();
+					ck = cache.generateCacheKey( entry.getId(), persister, source.getFactory(), source.getTenantIdentifier() );
+					lock = cache.lockItem( source, ck, entry.getVersion() );
+				}
+
+				if ( persister.isVersioned() && requestedLockMode == LockMode.PESSIMISTIC_FORCE_INCREMENT  ) {
+					// todo : should we check the current isolation mode explicitly?
+					Object nextVersion = persister.forceVersionIncrement( entry.getId(), entry.getVersion(), source );
+					entry.forceLocked( object, nextVersion );
+				}
+				else {
+					( (ReactiveEntityPersister) persister )
+							.reactiveLock( entry.getId(), entry.getVersion(), object, lockOptions, source );
```

Comment:
Ufffff, this isn't right.

Did I really write that?! Terrible.

Reply:
Just for my own knowledge, what's wrong about it?

Reply:
It just throws away the `CompletionStage`. It's never actually going to hit the database and lock the row.

Reply:
It's like the easiest and most basic error to make in writing reactive code.

Reply:
Totally

Reply:
Actually, it will hit the database, it's a `CompletionStage`, not a `Uni`, but it obviously won't be done in the proper sequence

Reply:
@geoand Are you still working on this? Otherwise we can merge it as it is and create a separate issue to fix this (we have the same bug in main anyway).

Reply:
Personally I'm in favor of getting parity with `main` as the failing test count is still high 🙂